content/articles/looking-at-payments.md

```diff
@@ -0,0 +1,100 @@
+
+If the customer is a business and is VAT registered, just getting the VAT number and checking it should be enough. However, companies paying without giving their VAT number or non-VAT registered ones should be treated as B2C.
+
+As mentioned in the introduction, we need to charge the VAT of the customer which means we need to locate it.
+
+Here are some examples of pieces of informations we can use:
+
+- billing address
+- ip address geolocation
+- bank location
+- card location
+
+Note that the legislation requires to have 2 non-conflicting informations and keep that for 10 (!) years. 
+If you don't manage that, you are supposed to reject the transaction. You could possibly ask for the user to send some proof by mail but that's a terrible user experience.
+
+If we manage to get customers to pay us, we then need to separate sales by country and apply the right VAT. This gives us the amount of tax we need to pay for each country.
+
+# Coding
+Now that we have an overview of what the VAT system looks like, let's have a look at the solutions.
+
+## Payment providers
+None of these actually help with the VAT mess but are still needed.
+There are a few alternatives here:
+
+- [Stripe](https://stripe.com)
+- [Braintree](https://www.braintreepayments.com/)
+- [Paymill](https://www.paymill.com/)
+
+Paymill is based in Europe. Stripe is extremely easy to setup and get started. I haven't used the other two so I can't comment on those but it seems you need to be approved for a merchant account for them which means some some wait.
+
+Here are the advertised rates (you can possibly get a better one by talking with them):
+- Paymill: 2.95% + 0.28€
+- Braintree: 2.4% + 20p (first £30k free) for EU cards, 3.4% + 20p for non EU cards
+- Stripe: 1.9% + 20p for UK cards and 2.9% + 20p for foreign cards
+
+Let's use a small example to see the difference: £10k sales with £5k from the UK, £3k from NA and £2k from Europe with let's say 100 transactions
+
+```python
+paymill = (2.95 * 10000 / 100) - (0.20 * 100)
+>> 275
+braintree = (2.4 * 7000 / 100) + (3.4 * 3000 / 100) - (0.20 * 100)
+>> 250
+stripe = (1.9 * 5000 / 100) + (2.9 * 5000 / 100) - (0.20 * 100)
+>> 220
+```
+
+Stripe seems to give the best price but keep in mind you cand probably get better rates than the advertised one (at least for Paymill according to the employee I met ages ago).
+
+## Handling VAT
+
+### Implementing it yourself
+You need to handle a few cases:
+
+#### Customer is VAT registered
+Ask for VAT number and use your country VAT is customer is based in your country or no VAT otherwise
+
+#### Customer doesn't provide VAT number
+Ask for country (that can be prefilled based on IP address) and compare with the IP location. If it matches continue otherwise compare the country filled with the bank country. If it matches continue otherwise ask the customer to send a fax with their certificate of birth and a passport. More seriously what should be done in the last case? 
```

Comment:
* Ask for "a" country
* the country provided (not filled)

More seriously: what  ...